src/main/java/org/broadinstitute/hellbender/cmdline/DeprecatedArtifactsRegistry.java

```diff
@@ -0,0 +1,57 @@
+package org.broadinstitute.hellbender.cmdline;
+
+import scala.Tuple2;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * When a tool is removed from GATK (after having been @Deprecated for a suitable period), an entry should
+ * be added to this list to issue a message when the user tries to run that tool.
+ *
+ * NOTE: Picard tools should be listed here as well, since by definition such tools will not be found in
+ * the Picard jar.
+ */
+public class DeprecatedArtifactsRegistry {
+
+    // Mapping from tool name to string describing the major version number where the tool first disappeared and
+    // optional recommended alternatives
+    private static Map<String, Tuple2<String, String>> deprecatedTools = new HashMap<>();
+
+    static {
+        // Indicate version in which the tool disappeared, and recommended replacement in parentheses if applicable
+        deprecatedTools.put("IndelRealigner", new Tuple2<>("4.0.0.0", "Please use GATK3 to run this tool"));
+        deprecatedTools.put("RealignerTargetCreator", new Tuple2<>("4.0.0.0", "Please use GATK3 to run this tool"));
```

Comment:
If these tools are going to be included after I implement them, saying that they are deprecated might be a strong statement: maybe a distinction between "not supported yet (but probably future support)" and "deprecated" is a better idea. Otherwise, if they are included again after version x.x.x.x, this message in previous version will be weird.

An idea might be to include a placeholder command line programs that just throw an "unsupported in the current version" for that kind of tools and annotated with `@ExperimentalFeature`, for example. Anyway, I hope to have `IndelRealignment` sooner than later...

Reply:
I think these messages should reflect the current state, rather than speculate about the future. Hopefully we can get these tools ported; at that point we can remove them from this list.

Reply:
Fair enough.